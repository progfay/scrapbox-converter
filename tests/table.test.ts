/* global describe it expect */

import { parse } from '@progfay/scrapbox-parser'
import convert from '../src'
import pretty from 'pretty'

describe('table', () => {
  it('Simple table', () => {
    const input = `Table
table:hello
  1	2	3
  1 	2 	3
  ------	------	------
  a	b	c`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })
})
