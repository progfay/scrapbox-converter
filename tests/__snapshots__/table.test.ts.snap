// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table Simple table 1`] = `
"<article>
  <h1 class=\\"title\\">Table</h1>
  <section>
    <table>
      <tr>
        <th>&nbsp;1</th>
        <th>2</th>
        <th>3</th>
      </tr>
      <tr>
        <td>&nbsp;1&nbsp;</td>
        <td>2&nbsp;</td>
        <td>3</td>
      </tr>
      <tr>
        <td>&nbsp;------</td>
        <td>------</td>
        <td>------</td>
      </tr>
      <tr>
        <td>&nbsp;a</td>
        <td>b</td>
        <td>c</td>
      </tr>
    </table>
  </section>
</article>"
`;
